### ISSUE TITLE
Sketch keyboard shortcut / insert link

### ISSUE THREAD

**Issue by jumpinggoofballs** [fabiocolacio/Marker#174]
The shortcut for the Sketch, `Ctrl+K`, is the universal shortcut for Insert Link - you can even see it here in the GitHub markdown editor. 

Could we have `Ctrl+K` for Insert Link as normal?

If so, then there are two issues: 
1. What should the behaviour be? I guess the GitHub way of doing things is fine:
a. no text selected: fills in `[<cursor>](url)`
b. random bit of text selected: fills in `[text](<cursor>)`
c. url selected: fills in `[<cursor>](https://github.com/)`
d. _optional_: if preceded by `!` to denote an image, I don't know whether people would want the cursor to go in the square brackets `![<cursor>](url)` in order to add a label / tag to the image as well, or whether the cursor should jump straight to the url like `![](<cursor>)` and leave the label / tag empty by default.
2. What should the alternative shortcut for Sketch be? This is a novel feature, so entirely up to you. But I guess something like `Ctrl+D` for 'draw' might make sense. Or really, anything.

PS: Can we have a new release soon? I would like all the new goodies from the last commits in the AUR. Thank you!

**Comment by Mandarancio** [comment]
Very good suggestion, I used `Ctrl + K` as `S` for `Sketch` was already taken, but `Ctrl + D` as draw make sense and if the `K` is already a standard in other editor it definitely make sens to use it for links. The github behaviours make sense I will implement it as soon as possible.

Thank you! :+1:

**Comment by Mandarancio** [comment]
The functionalities have been added without the (d) behaviour for the moment in commit 5b534ba. 

@jumpinggoofballs  let me know if it works as supposed.
**please** remember to update the git sub-module as well: `git submodule update --recursive`

**Comment by jumpinggoofballs** [comment]
Thanks!

Behaviour `c` doesn't work quite like the GitHub editor though. Instead it treats the URL like ordinary text and performs behaviour `b`.

BUT: I have not done `$ sudo ninja install` because I have the last release installed from the AUR and I don't want to fry it. Instead I have run it from the build folder, and a bunch of other stuff that links to external libraries isn't quite working either. So it might be just that. 

Is there a way to run the latest git master version properly while also having the AUR release installed? If so, I will be happy to keep testing and suggesting functionality :smile:

**Comment by Mandarancio** [comment]
It is not this one the behaviour (c): 

![maker_autolink](https://user-images.githubusercontent.com/276686/37863213-d477857e-2f5a-11e8-8003-ff45e124a285.gif)

It doesn't work like this on your setup? 

To install both `AUR` and git version is not that simple, however I think that the `AUR` package install marker in `/usr/` while by default ninja install it in `/usr/local/`, if you do not have `/usr/local/bin/` in your 
path you will have to type every time the full path to the executable, or you could create a link in the `/usr/bin/` something like:

```
ln -s /usr/local/bin/marker /usr/bin/marker_git
```

However I'm not 100% sure that there will not any conflict (e.g., what will happen to the icon louncher? It will be replaced by the git local installation?). Let me know what you think about this solution and its limitation.

**Comment by jumpinggoofballs** [comment]
Okay, so in the interest of being useful, I removed the AUR version and am running just the git master branch. 

**Observations:**

- the program installs in `/usr/share` from the AUR and `/usr/share/local` with ninja.

- behaviour `c` does not, unfortunately work on my setup:

![screenshot from 2018-03-24 12-39-43](https://user-images.githubusercontent.com/26354453/37864033-a2ef0120-2f60-11e8-8f0f-0d8a9d930a6f.png)
`CTRL+K =>`
![screenshot from 2018-03-24 12-39-56](https://user-images.githubusercontent.com/26354453/37864035-a73f9a0a-2f60-11e8-8b0b-1d76dd364ac9.png)


How does one remove a ninja installed package cleanly? Do I have to manually `rm` things?

**Comment by Mandarancio** [comment]
Ok the problem is that I did not test the ```http://....``` url, and this actually do not work. I will fix it!

To remove simply:
```bash
cd build
sudo ninja uninstall

```

**Comment by Mandarancio** [comment]
With commit  7edbe4f both safe link (e.g. `http://www.....`) and email are correctly identified as link and behaviour (c) should fully work.

**Comment by jumpinggoofballs** [comment]
Can confirm `http://www...` works.

However, something like just `google.com` does not. I think it probably should, no?

For what it's worth, I found this: [http://urlregex.com/](http://urlregex.com/) - the regex for the full standard complient URL format in a variety of languages. Hope it helps with the URL check.

**Comment by Mandarancio** [comment]
Perfect! 

For the ```xyz.com``` behaviour I'm not sure as originally `hoedown` and so now `scidown` do not support auto-link for this kind of url and not even the github editor does (nor for the shortcut), so I think there is some reason behind, maybe?

**Comment by jumpinggoofballs** [comment]
I'm guessing it's to do with the fact that in the context of writing code, you would have many occurrences of `text.text`. You would not want to automatically assume that that thing is a link.

_However_, in the context of an _Insert Link_ shortcut, you know for a fact that that's supposed to be a URL :stuck_out_tongue: 

In general, I think in other text editors the behaviour you would have there is that the checkIfURL looks at something like `google.com`, since you are trying to insert a link it figures that that is a valid URL, and then it would complete it to something like `[google.com](http://google.com)` - it would even add the `http(s)://` for the link. 

But I can also see the logic of leaving things as they are now. That kind of behaviour needs to make perhaps too many assumptions. I think inferring that that is a URL since you are trying to Insert a Link makes sense. But then when you fill it the protocol do you do `http://` or do you do `https://`? Either can be problematic as a default assumption.
